src/metabase/util/honey_sql_2_extensions.clj

```diff
@@ -0,0 +1,356 @@
+  see [[metabase.util.honeysql-extensions]], which at the time of this writing still uses Honey SQL 1."
+  (:refer-clojure
+   :exclude
+   [+ - / * abs mod inc dec cast concat format second])
+  (:require
+   [clojure.string :as str]
+   [honey.sql :as sql]
+   [metabase.util :as u]
+   [metabase.util.malli :as mu]
+   [metabase.util.schema :as su]
+   [potemkin.types :as p.types])
+  (:import
+   (java.util Locale)))
+
+(defn- english-upper-case
+  "Use this function when you need to upper-case an identifier or table name. Similar to `clojure.string/upper-case`
+  but always converts the string to upper-case characters in the English locale. Using `clojure.string/upper-case` for
+  table names, like we are using below in the `:h2` `honeysql.format` function can cause issues when the user has
+  changed the locale to a language that has different upper-case characters. Turkish is one example, where `i` gets
+  converted to `İ`. This causes the `SETTING` table to become the `SETTİNG` table, which doesn't exist."
+  [^CharSequence s]
+  (-> s str (.toUpperCase Locale/ENGLISH)))
+
+(sql/register-dialect!
+ :h2
+ (update (sql/get-dialect :ansi) :quote (fn [quote]
+                                          (comp english-upper-case quote))))
+
+;; register the `extract` function with HoneySQL
+;; (hsql/format (sql/call :extract :a :b)) -> "extract(a from b)"
+(defn- format-extract [_fn [unit expr]]
+  (let [[sql & args] (sql/format-expr expr)]
+    (into [(str "extract(" (name unit) " from " sql ")")]
+          args)))
+
+(sql/register-fn! ::extract format-extract)
+
+;; register the function `distinct-count` with HoneySQL
+(defn- format-distinct-count
+  "(sql/format-expr [::h2x/distinct-count :x])
+   =>
+   count(distinct x)"
+  [_fn [expr]]
+  (let [[sql & args] (sql/format-expr expr)]
+    (into [(str "count(distinct " sql ")")]
+          args)))
+
+(sql/register-fn! ::distinct-count format-distinct-count)
+
+;; register the function `percentile` with HoneySQL
+;; (hsql/format (sql/call :percentile-cont :a 0.9)) -> "percentile_cont(0.9) within group (order by a)"
+(defn- format-percentile-cont [_fn [expr p]]
+  {:pre [(number? p)]}
+  (let [[sql & args] (sql/format-expr expr)]
+    (into [(str "PERCENTILE_CONT(" p ") within group (order by " sql ")")]
+          args)))
+
+(sql/register-fn! ::percentile-cont format-percentile-cont)
+
+;; HoneySQL 0.7.0+ parameterizes numbers to fix issues with NaN and infinity -- see
+;; https://github.com/jkk/honeysql/pull/122. However, this broke some of Metabase's behavior, specifically queries
+;; with calculated columns with numeric literals -- some SQL databases can't recognize that a calculated field in a
+;; SELECT clause and a GROUP BY clause is the same thing if the calculation involves parameters. Go ahead an use the
+;; old behavior so we can keep our HoneySQL dependency up to date.
+#_(extend-protocol honeysql.format/ToSql
```

Comment:
why is this commented out?

Reply:
there's no equivalent way to do it in Honey SQL 2 -- I commented it out initially until I could find one. But since it's not possible I'll just remove the code. I forgot to do that here